socketio_middleware/src/rooms.rs

```diff
@@ -33,35 +35,114 @@ impl ChannelPair {
 }
 
 pub fn join_channel_to_room(room_id: &str, channel_pair: ChannelPair) {
+    //TODO: why use remove, it is low performance.
     let mut connected_sockets = match ROOMS.remove(room_id) {
         Some(val) => val,
         None => Vec::new(),
     };
 
-    connected_sockets.push(channel_pair);
+    //check if socketid exist
+    let mut exist: bool = false;
```

Comment:
nit: remove `bool`, as the compiler knows what it is by definition, and should be `exists`

Reply:
sure